api/endpoints/stocks.py

```diff
@@ -1,8 +1,14 @@
-from fastapi import APIRouter
+from fastapi import APIRouter, Depends
+from starlette import status
+
+from api.deps import get_data_providers
+from remote.data_provider.data_provider import DataProvider
+from schemas.stock_price_information import StockPriceInformation
+from services import stock_data_service
 
 router = APIRouter()
 
 
-@router.get("/stocks/{name}/price")
-async def get_stock_quotes(name: str):
-    return {"message": f"Hello {name}"}
+@router.get("/stocks/{ticker}/quotes", response_model=set[StockPriceInformation], status_code=status.HTTP_200_OK)
```

Comment:
> ```python
> Depends(get_data_providers)):
> ```

This lines makes use of FastAPI's strong [Dependency Injection](https://fastapi.tiangolo.com/tutorial/dependencies/#dependencies-first-steps)